datafusion/expr/src/utils.rs

```diff
@@ -379,14 +379,12 @@ fn get_exprs_except_skipped(
     }
 }
 
-/// Resolves an `Expr::Wildcard` to a collection of `Expr::Column`'s.
-pub fn expand_wildcard(
-    schema: &DFSchema,
-    plan: &LogicalPlan,
-    wildcard_options: Option<&WildcardOptions>,
-) -> Result<Vec<Expr>> {
+/// For each column specified in the USING JOIN condition, the JOIN plan outputs it twice
+/// (once for each join side), but an unqualified wildcard should include it only once.
+/// This function returns the columns that should be excluded.
+fn exclude_using_columns(plan: &LogicalPlan) -> Result<HashSet<Column>> {
```

Comment:
This function is extracted from `expand_wildcard`, so that we can reuse it in `exprlist_to_fields`.

Reply:
Although we have moved wildcard expansions to the analyzer #11681, it still does wildcard expansions when computing plan schemas(in [exprlist_to_fields](https://github.com/apache/datafusion/blob/f9cc3325cdb5891b7566a6f3503c1f7ac6ad51e0/datafusion/expr/src/utils.rs#L696) and [exprlist_len](https://github.com/apache/datafusion/blob/f9cc3325cdb5891b7566a6f3503c1f7ac6ad51e0/datafusion/expr/src/utils.rs#L797C8-L797C20)).  I wonder if performing wildcard expansions before computing schemas would be simplier, at least it would avoid redundant work.